protoc-gen-twirp/generator.go

```diff
@@ -1334,8 +1334,15 @@ func (t *twirp) generateServerProtobufMethod(service *descriptor.ServiceDescript
 	t.P()
 	t.P(`  buf, err := `, t.pkgs["ioutil"], `.ReadAll(req.Body)`)
 	t.P(`  if err != nil {`)
-	t.P(`    s.writeError(ctx, resp, wrapInternal(err, "failed to read request body"))`)
-	t.P(`    return`)
+	t.P(`  		if ctxErr := context.Canceled; ctxErr == ctx.Err() {`)
+	t.P(`    		s.writeError(ctx, resp, twirp.NewError(twirp.Canceled, ctxErr.Error()))`)
+	t.P(`    		return`)
+	t.P(`  		} else if ctxErr := context.DeadlineExceeded; ctxErr == ctx.Err() {`)
+	t.P(`    		s.writeError(ctx, resp, twirp.NewError(twirp.DeadlineExceeded, ctxErr.Error()))`)
+	t.P(`    		return`)
+	t.P(`  		}`)
+	t.P(`    	s.writeError(ctx, resp, twirp.NewError(twirp.Malformed, "failed to read request body"))`)
```

Comment:
A couple lines down, this function uses the `malformedRequestError` helper to get a similar result in a different case. Please make both/all call sites look the same: this line should use the same helper, or all lines like that one should avoid it. (I don't know exactly why that helper exists.)